liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/PluginConfigXmlDocument.java

```diff
@@ -0,0 +1,191 @@
+        }
+    }
+    
+    public void createElement(String name, Map<String, String> values) {
+        if (values == null || values.isEmpty()) {
+            return;
+        }
+        Element child = doc.createElement(name);
+        for (Map.Entry<String, String> entry : values.entrySet()) {
+            createElement(child, entry.getKey(), entry.getValue());
+        }
+        doc.getDocumentElement().appendChild(child);
+    }
+    
+    public void createElement(String name, List<String> values) {
+        if (values == null || values.isEmpty()) {
+            return;
+        }
+        Element child = doc.createElement(name);
+        for (int i = 0; i < values.size(); i++) {
+            createElement(child, "param", values.get(i));
+        }
+        doc.getDocumentElement().appendChild(child);
+    }
+    
+    public void createElement(String name, ArtifactItem value) {
+        if (value == null) {
+            return;
+        }
+        Element child = doc.createElement(name);
+        createElement(child, "groupId", value.getGroupId());
+        createElement(child, "artifactId", value.getArtifactId());
+        createElement(child, "version", value.getVersion());
+        createElement(child, "version", value.getType());
+        doc.getDocumentElement().appendChild(child);
+    }
+    
+    public void createElement(String name, Install value) {
+        if (value == null) {
+            return;
+        }
+        Element child = doc.createElement(name);
+        
+        if (value.getCacheDirectory() != null) {
+            createElement(child, "cacheDirectory", value.getCacheDirectory());
+        }
+        if (value.getLicenseCode() != null) {
+            createElement(child, "licenseCode", value.getLicenseCode());
+        }
+        if (value.getType() != null) {
+            createElement(child, "type", value.getType());
+        }
+        if (value.getType() != null) {
+            createElement(child, "type", value.getType());
+        }
+        if (value.getRuntimeUrl() != null) {
+            createElement(child, "runtimeUrl", value.getRuntimeUrl());
+        }
+        if (value.getUsername() != null) {
+            createElement(child, "username", value.getUsername());
+        }
+        if (value.getPassword() != null) {
+            createElement(child, "password", "*********");
+        }
+        createElement(child, "maxDownloadTime", Long.toString(value.getMaxDownloadTime()));
+        if (value.getRuntimeUrl() != null) {
+            createElement(child, "runtimeUrl", value.getRuntimeUrl());
+        }
+        createElement(child, "type", value.isVerbose());
+        doc.getDocumentElement().appendChild(child);
+    }
+ 
+    public void createElement(String key, String value) {
+        Element child = doc.createElement(key);
+        child.appendChild(doc.createTextNode(value));
+        doc.getDocumentElement().appendChild(child);
```

Comment:
this could replace 147-149 with
createElement(doc.getDocumentElement, key, value);